File: Army/units/soldier.cpp
#include "soldier.h"


using namespace std;

Soldier::Soldier(const string& name, int healthPoint, int damage) : Unit(name, healthPoint, damage) {
    setAbility(new DefaultAbility(this));
    setCurrentState(new HumanState(this));
    setNextState(new HumanState(this));
    setUnitType(soldier);
}

Soldier::~Soldier() {}

File: Java/src/com/gymfox/army/units/Soldier.java
package com.gymfox.army.Units;

import com.gymfox.army.Ability.DefaultAbility;

public class Soldier extends Unit {
    public Soldier(String name, int healthPointLimit, int damage) {
        super(name, healthPointLimit, damage);
        this.ability = new DefaultAbility(this);
    }
}
